coordinator/src/escrow_coordinator/mod.rs

```diff
@@ -29,52 +31,66 @@ impl EscrowCoordinator {
-                        if self.pending_contracts.contains_key(&contract_hash) {
-                            self.pending_contracts.remove(&contract_hash);
-                            self.begin_trade(&contract_hash, &contract).await?;
-                        } else {
-                            self.pending_contracts.insert(contract_hash, contract);
+        loop {
+            match notifications.recv().await {
+                Ok(notification) => {
+                    if let RelayPoolNotification::Event { event, .. } = notification {
+                        if let Ok(unwrapped_gift) =
+                            self.nostr_client.client.unwrap_gift_wrap(&event).await
+                        {
+                            let rumor = unwrapped_gift.rumor;
+                            if rumor.kind == Kind::PrivateDirectMessage {
+                                if let Ok((contract_hash, contract)) =
+                                    EscrowCoordinator::parse_contract(&rumor.content)
+                                {
+                                    dbg!("Received contract: {}", &contract.trade_description);
+                                    if self.pending_contracts.contains_key(&contract_hash) {
+                                        self.pending_contracts.remove(&contract_hash);
+                                        let _ = self
+                                            .begin_trade(&contract_hash, &contract)
+                                            .await
+                                            .inspect_err(|e| {
+                                                //todo: use logger instead
+                                                println!(
+                                                    "Got error while beginning a trade: {}",
+                                                    e
+                                                );
+                                            });
+                                    } else {
+                                        self.pending_contracts.insert(contract_hash, contract);
+                                    }
+                                }
+                            }
                         }
+                    } else if RelayPoolNotification::Shutdown == notification {
+                        break Err(anyhow!(
+                            "Got shutdown notification, breaking coordinator lopp!"
+                        ));
                     }
                 }
+                Err(RecvError::Closed) => {
+                    break Err(anyhow!(
+                        "Got closed error from channel, breaking coordinator lopp!"
+                    ))
+                }
+                Err(RecvError::Lagged(count)) => {
+                    //todo: use logger instead
+                    println!("Lost {} events, resuming after that...", count);
```

Comment:
```suggestion
                    eprintln!("Lost {} events, resuming after that...", count);
```
Either we use logging here too (as the comment indicates), or i think it would make sense to print errors to stderr using eprintln! instead of println!
```suggestion
                    println!("Lost {} events, resuming after that...", count);
```

Reply:
It sound as a quick improvement.